src/components/tables/DataTable.js

```diff
@@ -177,56 +177,59 @@ export default {
     search () {
       this.page = 1
     },
-    page () {
-      this.update()
+    filteredItems: {
+      handler: function () {
+        if (this.everyItem) this.all = true
+      },
+      deep: true
+    },
+    selected () {
+      if (this.everyItem) this.all = true
     }
   },
 
   methods: {
-    update () {
-      this.$emit('update', {
-        page: this.page,
-        rowsPerPage: this.rowsPerPage,
-        sorting: this.sorting,
-        desc: this.desc
-      })
+    isSelected (item) {
+      return this.selected && this.selected.find(i => i[this.selectedKey] === item[this.selectedKey])
     },
     sort (index) {
-      if (this.sorting === null) {
-        this.sorting = index
-        this.desc = false
-      } else if (this.sorting === index && !this.desc) {
-        this.desc = true
-      } else if (this.sorting !== index) {
-        this.sorting = index
-        this.desc = false
+      if (this.pagination.sortBy === null) {
+        this.$emit('update:pagination', Object.assign({}, this.pagination, { sortBy: index, descending: false }))
+      } else if (this.pagination.sortBy === index && !this.pagination.descending) {
+        this.$emit('update:pagination', Object.assign({}, this.pagination, { descending: true }))
+      } else if (this.pagination.sortBy !== index) {
+        this.$emit('update:pagination', Object.assign({}, this.pagination, { sortBy: index, descending: false }))
       } else {
-        this.sorting = null
-        this.desc = null
+        this.$emit('update:pagination', Object.assign({}, this.pagination, { sortBy: null, descending: null }))
       }
-
-      this.update()
     },
     genTR (children, data = {}) {
       return this.$createElement('tr', data, children)
     },
-    toggle (val) {
-      this.all = val
+    toggle (value) {
+      this.all = value
 
-      this.$emit('input', this.value.map(i => {
-        i[this.itemValue] = this.filteredItems.includes(i) ? val : false
+      let selected = this.selected.slice()
+      if (value)
+        selected.push(...this.filteredItems)
+      else
+        selected = selected.filter(i => !this.filteredItems.find(j => j[this.selectedKey] === i[this.selectedKey]))
 
-        return i
-      }))
+      this.$emit('update:selected', selected)
     }
   },
 
   mounted () {
-    const header = this.headers.find(h => !('sortable' in h) || h.sortable)
-    this.desc = this.defaultSort ? this.defaultSort.desc : this.desc
-    this.sorting = this.defaultSort ? this.defaultSort.field : header.value
+    const firstSortable = this.headers.find(h => !('sortable' in h) || h.sortable)
+
+    let defaultPagination = {
```

Comment:
should be const as it doesn't change value, could also just put it into the Object.assign arguments.